handout/testing/components/injecting-dependencies.md

```diff
@@ -46,39 +46,52 @@ class MockQuoteService {
   }
 }
 
-import {QuoteService} from './quote.service.ts';
-import {QuoteComponent} from './quote.component.ts';
+import {QuoteService} from 'quote.service.ts';
+import {QuoteComponent} from 'quote.component.ts';
+import {provide} from '@angular/core';
 import {
+  async,
   expect,
   it,
   describe,
-  async
-  TestComponentBuilder,
-  beforeEachProvider
+  inject,
+  beforeEach,
+  beforeEachProvider,
 } from '@angular/core/testing';
 
+import {
+  TestComponentBuilder,
+  ComponentFixture,
+}
+from '@angular/compiler/testing';
+
 describe('Testing Quote Component', () => {
-  beforeEachProviders(() => {
-    provide(QuoteService: {useClass: MockQuoteService})
-  });
 
-  it('Should get quote', async(inject([TestComponentBuilder], (tcb) => {
-    return tcb.createAsync(QuoteComponent).then(fixture) => {
+  beforeEachProviders(() => [
+    provide(QuoteService, { useClass: MockQuoteService })
+  ]);
+
+  it('Should get quote',
+    async(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+      tcb.createAsync(QuoteComponent).then(fixture => {
 
-      fixture.debugElement.componentInstance.getQuote();
+        fixture.componentInstance.getQuote();
 
-      fixture.detectChanges();
-      var compiled = fixture.debugElement.nativeElement;
-      expect(compiled.querySelector('div')).toHaveText('Test Quote');
-    }
-  })));
+        fixture.whenStable().then(() => {
```

Comment:
`whenStable` ensure the promise completes -- this allows us to test async actions when we don't know the amount of time that needs to pass.